collect_app/src/main/java/org/odk/collect/android/http/HttpInterface.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.http;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.odk.collect.android.utilities.ResponseMessageParser;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+public interface HttpInterface {
+
+    /**
+     * Creates a http connection and sets up an input stream.
+     *
+     * @param uri of the stream
+     * @param contentType check the returned Mime Type to ensure it matches. "text/xml" causes a Hash to be calculated
+     * @return HttpInputStreamResult - An object containing the Stream, Hash and Headers
+     * @throws Exception a multitude of Exceptions such as IOException can be thrown
+     */
+    @NonNull
+    HttpInputStreamResult getHttpInputStream(@NonNull URI uri, @Nullable String contentType) throws Exception;
+
+    /**
+     * Performs a HTTP Head request.
+     *
+     * @param uri of which to perform a HTTP head
+     * @param responseHeaders Map which is populated with the HTTP Headers
+     * @return HTTP status code
+     * @throws Exception a multitude of Exceptions such as IOException can be thrown
+     */
+    int httpHeadRequest(@NonNull URI uri, @NonNull Map<String, String> responseHeaders) throws Exception;
```

Comment:
Here we could avoid having that output arg by returning something similar to the HttpInputStreamResult. We could call it `HttpHeadResult` and it could have methods to know if the request was a success (2xx status code), if it was an HTTP 204 No Content, or an HTTP 401 Unauthorized.

Feel free to browse a branch I've pushed to my fork with the changes I made to the code while reviewing it https://github.com/ggalmazor/collect/commits/pr_2257_changes

Reply:
I've pushed an update, `httpHeadRequest()` now returns an instance of a new `class HttpHeadResult` which contains the headers + status code. As per comment about renaming `getHttpInputStream` I also renamed this method to `head`.

I've just added a getter for status code (At least for now), doing so doesn't require the logic of the callers to change too much.

Reply:
Great! :)